lib/util/component.js

```diff
@@ -13,14 +13,26 @@
  * @param {Context} context The Eslint context.
  */
 function isComponent(node, context) {
-    if (!node.superClass) {
+    const { superClass } = node;
+
+    if (!superClass) {
         return false;
     }
 
-    // Checking if the class extends `Element` covers 99% of the LWC cases. We can make this safer
-    // in the future by ensuring that the `superClass` identifier is imported from engine.
-    const sourceCode = context.getSourceCode();
-    return sourceCode.getText(node.superClass) === 'LightningElement';
+    const program = context.getAncestors(node).find(({ type }) => type === 'Program');
+
+    const importDeclaration = program.body
+        .filter(({ type }) => type === 'ImportDeclaration')
+        .find(({ source }) => source.value === 'lwc');
```

Comment:
would this do the same?

```suggestion
        .find(({ source, type }) => type === 'ImportDeclaration' && source.value === 'lwc');
```